### ISSUE TITLE
Feature request: Icons inside BarChart and PieChart

### ISSUE THREAD

**Issue by georgekal2798** [Abhinandan-Kushwaha/react-native-gifted-charts#500]
Hello there!

I would like to request a feature, that was inspired by the specs of an app I am developing.

Inside the bars of a BarChart and the slices of a PieChart (in this case, donut), I need to show an icon, something like this.
<img width="278" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/38210685/0d56348b-6f3b-48d7-bcc7-baeabbb4d323">
<img width="270" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/38210685/1670b947-db90-44f9-bea9-9e5675a3919d">

I know in the PieChart it is possible to show text, but it is not as customisable as I would want to. I tried to make it work with a custom icon font, but I can't change the font family of the text.
<img width="284" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/38210685/4ee6b9d4-cccb-4469-a9b8-b7ac04b546f2">
Also, with the BarChart, I was able to show the icons using the font family and I display the respective icons as the x axis labels, but again this is not optimal, as I can't recreate the round white background.
<img width="178" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/38210685/bd3c9205-7ece-4c0e-b379-3ba8232143ff">

I would really appreciate it if this could be made into a feature: either placing a custom element in the bar/slices, or being able to further customise the text and the background. Please let me know if this is possible and how much time it would take, because I have to know if I can implement it like this in the coming days. Also, if I have missed something and this is feasible using the current version, please tell me how to do it.

Thank you very much in advance!

**Comment by Abhinandan-Kushwaha** [comment]
Hi @georgekal2798 👋
Thanks for requesting this feature. I will look into this and come up with a solution by tomorrow.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @georgekal2798 👋
This feature is now available from version `1.3.33` onwards.🎉

Added 2 new props- `barInnerComponent` and `pieInnerComponent`.
Both of these are callback functions that accept 2 parameters - `item` and `index`  and should return a UI element (React node)

For Bar charts, the UI element is rendered inside the bars. It is positioned at the top of the Bar by default. It can be repositioned using CSS techniques.

Here's an example-
```js
const barData = [
  {value: 15},
  {value: 30},
  {
    value: 26,
    barInnerComponent: (item, index) => {
      return (
        <View
          style={{
            flex: 1,
            alignItems: 'center',
            backgroundColor: 'green',
          }}>
          <View
            style={{
              height: 20,
              width: 20,
              borderRadius: 10,
              backgroundColor: 'red',
            }}>
            <Text style={{color: 'blue', textAlign: 'center'}}>
              {item.value}
            </Text>
          </View>
        </View>
      );
    },
  },
  {value: 40},
];
return (
  <BarChart
    data={barData}
    barInnerComponent={(item, index) => {
      return (
        <View
          style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
          <View
            style={{
              height: 20,
              width: 20,
              borderRadius: 10,
              backgroundColor: 'yellow',
            }}>
            <Text style={{color: 'blue', textAlign: 'center'}}>{item.value}</Text>
          </View>
        </View>
      );
    }}
  />
);
```
The output of the above code is-

<img width="242" alt="Screenshot 2024-01-19 at 5 09 07 AM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/c7cd8405-9839-4c1e-a2b9-33eae76e4053">


For Pie and Donut charts, the `pieInnerComponent` is a callback function that should return a `svg` element imported from `react-native-svg`.
Here's an example of `pieInnerComponent` -
```js
pieInnerComponent={(item,index)=><SvgText x={0} y={0} height={20} width={40} fill={'red'} stroke={'blue'}>
  {item.value}
</SvgText>}
```

**Note**- Both `barInnerComponent` and `pieInnerComponent` can be also be added as a property to the data objects.

**Comment by georgekal2798** [comment]
WOW! Thank you for your fast response! Amazing work! Keep it up 👍

**Comment by georgekal2798** [comment]
I managed to make it work but I faced a bug when I enabled the focusOnPress property: When the slice is expanded, the inner component is now hidden
<img width="296" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/38210685/253e1ad8-a6f4-4b79-92c8-514ee2aec7cc">
<img width="302" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/38210685/8bcc6f39-0b2c-4c91-8b0e-555d6037c6bc">

**Comment by Abhinandan-Kushwaha** [comment]
Hi @georgekal2798 Can you share the code?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @ahsan85 
Can you please share the screenshots of the current output and the desired output

**Comment by Abhinandan-Kushwaha** [comment]
Hi @ahsan85 👋
Thanks for reporting this issue. I will release a new version with the fix today.

For now, I can suggest you a workaround. The `barInnerComponent` is currently working fine for gradient bar charts. Just pass the prop `showGradient`. Additionally, you should pass the prop `gradientColor={colors.primary}` to remove the gradient effect from bars.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @ahsan85 
This has been fixed in version `1.4.5` 
Please use the latest version of the library.
